src/azure-cli/azure/cli/command_modules/rdbms/custom.py

```diff
@@ -698,13 +829,15 @@ def create_database(cmd, resource_group_name, server_name, database_name, engine
     if engine_name == 'mysql':
         # check for existing database, create if not present
         database_client = cf_mysql_db(cmd.cli_ctx, None)
+        parameters = mysql.models.Database(name=database_name, charset='utf8')
     elif engine_name == 'mariadb':
         database_client = cf_mariadb_db(cmd.cli_ctx, None)
+        parameters = mariadb.models.Database(name=database_name, charset='utf8')
     try:
-        database_client.get(resource_group_name, server_name, database_name).result()
-    except CloudError:
+        database_client.get(resource_group_name, server_name, database_name)
+    except ResourceNotFoundError:
         logger.warning('Creating %s database \'%s\'...', engine_name, database_name)
-        database_client.create_or_update(resource_group_name, server_name, database_name, 'utf8').result()
+        database_client.begin_create_or_update(resource_group_name, server_name, database_name, parameters)
```

Comment:
`create_or_update` and `begin_create_or_update` both return `LROPoller`, why drop `.result()` in latest code?